lib/DDGC/Util/Script/Docs.pm

```diff
@@ -0,0 +1,283 @@
+            ], \$in, \$out, \$err, timeout(60) or die "$err (error $?) $out";
+        }
+    }
+
+    return $source_dir;
+}
+
+# Builds the prev/next hash to push to each template.
+sub get_prev_next {
+    my $self = shift;
+
+    my $prev_next_path = dir($self->source_dir,$self->prev_next_file);
+    my $json = io($prev_next_path)->slurp;
+    $json =~ s/\.md//g;
+    my $data = decode_json $json;
+
+    return $data;
+}
+
+
+sub get_nav {
+    my $self = shift;
+
+    my $nav_path = dir( $self->source_dir, $self->nav_file );
+    my $markdown = io($nav_path)->slurp;
+
+    my %nav = ();
+    my @nav = ();
+  HEADING: foreach my $heading ( split( /\n\-/s, $markdown ) ) {
+        next HEADING if !$heading;
+        next HEADING if $heading !~ /\*\*/;
+
+        my $title = '';
+        my @sec   = ();
+      LINE: foreach my $line ( split( /\n/, $heading ) ) {
+            if ( $line =~ /\*\*([^\*]+)/ ) {
+                $title = $1;
+
+                # Capturing markdown link.
+            }
+            elsif ( $line =~ /\[([^\]]+)\]\(([^\)]+)/ ) {
+                my $section = $1;
+                my $link    = $2;
+                $link =~ s/^.*\/(.*)\.md/$1/;
+                my %sec = ();
+                $sec{'title'} = $section;
+                $sec{'link'}  = $link;
+                push( @sec, \%sec );
+
+                $nav{$link} = {
+                    'category' => $title,
+                    'title'    => $section,
+                };
+            }
+        }
+
+        if ($title) {
+            my %sec = ();
+            $sec{'title'}    = $title;
+            $sec{'sections'} = \@sec;
+            push( @nav, \%sec );
+        }
+    }
+
+    return \@nav, \%nav;
+}
+
+sub pages {
+    my $self = shift;
+
+    my %pages = ();
+
+    my $prev_next_hash_ref = $self->get_prev_next;
+    my %prev_next          = %{$prev_next_hash_ref};
+
+    my ( $nav_arr_ref, $nav_hash_ref ) = $self->get_nav;
+    my %nav = %{$nav_hash_ref};
+
+    find(
+        sub {
+
+            my $name = $File::Find::name;
+            my $dir  = $File::Find::dir;
+
+            return if $name =~ /$self->{nav_file}/;
+
+            return unless $name =~ /^[^.].+\.md$/;    # only markdown files
+
+            my ($file) = $name =~ /^$self->{source_dir}\/(.*)\.md/;
+            my $dir_rel = '';
+            ( $dir_rel, $file ) = $file =~ /^(.*)\/(.*)/ if $file =~ /\//;
+
+            my $markdown = io($name)->slurp;
+
+# Replaces hard github links to other markdown files to our newly converted relative links
+            $markdown =~
+s~(\]\()https://github.com/duckduckgo/duckduckgo-documentation/blob/master/duckduckhack/(?:[^\/\.]+\/){1,4}([^\.]+?)\.md([^\)]*?\))~$1$2$3~sg;
+
+ # Replace code types the parser can't deal with,
+ # that is a #### line without a space line between it and a code start line ```
+            $markdown =~ s/(\n\#+.*?)(\n`)/$1\n$2/sg;
+
+            my $buffer = q{};
+            open my $fh, '>', \$buffer;
```

Comment:
`IO::All` supports in-memory file handles, but this code came straight from publisher and I'm not too interested right now in making changes to it.

It's essentially a black box for the moment.